tests/devices/_internal/test_file_parser.py

```diff
@@ -111,6 +123,43 @@ def test_none_if_not_found(self, tmp_path):
         assert read_device_files([tmp_path]).online_id is None
 
 
+class TestReadsJlinkData:
+    def test_reads_target_name(self, tmp_path):
+        board_name = "FRDM-K66F"
+        pathlib.Path(tmp_path, "Segger.html").write_text(build_segger_html())
+        file_contents = build_board_html(f"https://www.dummy.com/{board_name}")
+        pathlib.Path(tmp_path, "Board.html").write_text(file_contents)
+        online_id = read_device_files([tmp_path]).online_id
+        assert online_id.slug == board_name
+        assert online_id.target_type == "jlink"
+
+    def test_target_name_none_if_not_found(self, tmp_path):
+        pathlib.Path(tmp_path, "Segger.html").write_text(build_segger_html())
+        pathlib.Path(tmp_path, "Board.html").write_text("<html></html>")
+        assert read_device_files([tmp_path]).online_id is None
+
+    def test_reads_jlink_type(self, tmp_path):
+        j_link_type = "j-link-ob"
+        pathlib.Path(tmp_path, "Segger.html").write_text(build_segger_html(j_link_type))
+        assert read_device_files([tmp_path]).interface_details == {"Version": j_link_type}
+
+    def test_jlink_interface_empty_if_not_found(self, tmp_path):
+        pathlib.Path(tmp_path, "Segger.html").write_text("<html></html>")
+        pathlib.Path(tmp_path, "Board.html").write_text(build_board_html())
+
+        assert read_device_files([tmp_path]).interface_details == {}
+
+
+class TestSlugParser:
```

Comment:
Can we test this through the public interface to the module? I'd prefer we didn't test implementation details directly as it will make our code easier to change later.